src/framework/operations.hpp

```diff
@@ -288,6 +288,19 @@ inline Op make_unitary(const reg_t &qubits, cmatrix_t &&mat, std::string label =
   return op;
 }
 
+inline Op make_diagonal(const reg_t &qubits, const cvector_t &vec, std::string label = "") {
```

Comment:
Since vec is usually a temporary in the fusion code that uses this, you should make a move-semantics version as well with
```suggestion
inline Op make_diagonal(const reg_t &qubits, cvector_t &&vec, std::string label = "") {
  // same but with
  op.params = std::move(vec);